### PR TITLE
feat: Customise JITSI Url path 

### PR DESCRIPTION
**Pull request by chief-nerd**
## What does this PR do?

Allows the user to set a custom URL path with replacements insead of the UUID:

https://meet.mydomain.com/D64138E9-8912-47FD-8ED8-7A4673E7522F
can become:
https://meet.mydomain.com/30min-Meeting-with-Paul

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How should this be tested?

- [ ] Add Pattern e.g. {Title}-{uuid}
- [ ] Create Jitis Meeting

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

@hariombalhara  ... didn't want the work from the JITSI HostURL to be unused (https://github.com/calcom/cal.com/pull/8636)
... so here is the code in a separate PR for discussion if that is useful (I think it is)

### CODE REVIEW

**Comment by chief-nerd** [comment]
@hariombalhara  ... didn't want the work to be unused 
... so here is the code in a separate PR for discussion if that is useful (I think it is)

File: packages/app-store/jitsivideo/lib/VideoApiAdapter.ts [hunk 1]
```diff
@@ -11,19 +12,28 @@ const JitsiVideoApiAdapter = (): VideoApiAdapter => {
...
+      //Allows "/{Type}-with-{Attendees}" slug
+      const meetingID = meetingPattern
+        .replaceAll("{uuid}", uuidv4())
+        .replaceAll("{Title}", eventData.title)
+        .replaceAll("{Type}", eventData.type)
+        .replaceAll("{Attendees}", eventData.attendees.map((a) => a.name).join("-"))
+        .replaceAll("{Organizer}", eventData.organizer.name)
+        .replaceAll("{Location}", eventData.location || "")
+        .replaceAll("{Team}", eventData.team?.name || "")
+        .replaceAll(" ", "-"); //Last Rule! - Replace all blanks (%20) with dashes;
```
Comment (hariombalhara) [inline packages/app-store/jitsivideo/lib/VideoApiAdapter.ts:30]: Screenshot 2023-05-19 at 1 23 14 PM
I would prefer that you reuse these variables and the corresponding logic. That would make it consistent.

**Review by hariombalhara — DISMISSED** [review]
@PeerRich Blocking it till we address if we want to do it and if it's the right approach.

**Comment by chief-nerd** [comment]
> @PeerRich Blocking it till we address if we want to do it and if it's the right approach.

Ok. Is there a timeframe?

File: packages/app-store/jitsivideo/lib/VideoApiAdapter.ts (hunk 1 above)
Comment (PeerRich) [reply packages/app-store/jitsivideo/lib/VideoApiAdapter.ts:30]: yea reusing makes sense, tho to be honest jitsi is not a big application / not used much so I don't mind merging whatever works 🙏

Comment (chief-nerd) [reply packages/app-store/jitsivideo/lib/VideoApiAdapter.ts:30]: Alright @hariombalhara ... then let's do it? :)

**Comment by PeerRich** [comment]
@hariombalhara would you mind re-reviewing?

File: packages/app-store/jitsivideo/lib/VideoApiAdapter.ts (hunk 1 above)
Comment (hariombalhara) [reply packages/app-store/jitsivideo/lib/VideoApiAdapter.ts:30]: @chief-nerd Let's reuse the existing variable names and if something doesn't exist as a variable then create a new variable name.

I will review it again on Monday

**Comment by PeerRich** [comment]
seems like some lint issues @chief-nerd

**Review by hariombalhara — CHANGES_REQUESTED** [review]
We need to use the same variables `Event Type Title`, `Organizer`, `Scheduler`, `Location`

**Comment by chief-nerd** [comment]
> We need to use the same variables `Event Type Title`, `Organizer`, `Scheduler`, `Location`

Can do it tomorrow. (including the linting).

**Comment by chief-nerd** [comment]
@hariombalhara fixed the variable names and corrected the linting. Ready to merge.

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**